python/ray/train/lightning/lightning_utils.py

```diff
@@ -49,6 +50,36 @@ def get_worker_root_device():
         return devices
 
 
+@PublicAPI(stability="alpha")
+def get_devices() -> Optional[Union[List[int], str]]:
+    """Returns the parallel devices for Lightning Trainer on each Ray Train worker.
+
+    This method returns the list of CUDA device indexes of a GPU worker, and returns 
+    "auto" if called inside a CPU worker. Note that you can only call this method within 
+    the training function of the :class:`TorchTrainer <ray.train.torch.TorchTrainer>` class.
```

Comment:
We may want to consider no-op outside of `TorchTrainer` so that the user can run `train_loop_per_worker()` directly.

Reply:
From some testing, we can probably remove this api and let the users set `pl.Trainer(devices=-1)`.